### PR TITLE
fix: AWS checkpoint download from master

### PR DESCRIPTION
**Pull request by erikwilson**
## Description

Ensure that cli deployed AWS clusters have access to checkpoint downloads.

## Test Plan

- Deploy a determined aws cluster using this code change.
- Create a run which generates a checkpoint.
- Download the checkpoint through the server, ie: `det checkpoint download UUID --mode=master`

## Commentary (optional)

This PR addresses permissions in AWS Cloudformation templates.

## Checklist

- [X] Changes have been manually QA'd (simple aws deployments)
- [ ] User-facing API changes need the "User-facing API Change" label.
- [X] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

## Ticket

DET-8762

[DET-8762]: https://determinedai.atlassian.net/browse/DET-8762?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by keita-determined** [comment]
i think you need to assign someone instead of https://github.com/orgs/determined-ai/teams/docs cuz there's no one in there

File: harness/determined/deploy/aws/templates/efs.yaml [hunk 1]
```diff
@@ -730,6 +730,13 @@ Resources:
                   - cloudwatch:PutMetricData
                 Effect: Allow
                 Resource: "*"
+        - PolicyName: master-s3-policy
+          PolicyDocument:
+            Version: 2012-10-17
```
Comment (eecsliu) [inline harness/determined/deploy/aws/templates/efs.yaml:735]: will we ever have to bump this version or is it pretty much good?

**Review by eecsliu — APPROVED** [review]
👨‍🍳 😚 👌

**Comment by erikwilson** [comment]
Tested with https://github.com/determined-ai/determined/pull/5752

File: harness/determined/deploy/aws/templates/efs.yaml (hunk 1 above)
Comment (erikwilson) [reply harness/determined/deploy/aws/templates/efs.yaml:735]: This looks to still be the most recent version for policy document.

**Pull request merged**